executor/include/executor/default/omp_executor.hpp

```diff
@@ -50,8 +50,8 @@ struct omp_executor {
   template <typename F>
   void bulk_execute(F&& f, shape_type n) const {
 #ifdef _OPENMP
-  #pragma omp parallel for num_threads(n)
-    for (int i = 0; i < n; ++i) f(omp_get_thread_num());
+  #pragma omp parallel num_threads(n)
+    f(omp_get_thread_num());
```

Comment:
Just a minor comment: the upper limits of shape (`n` for OMP and `dim.{x,y,z}`) are needed for a lot of logic in the loops. How are we expecting these to be passed to the lambda? By the user, via the `shared_something` as in the pdf you linked to or we don't care?

Reply:
I am not quite sure about what the `SharedFactory` is supposed to do, and seems to be absent in the later versions.

Regarding the upper limit, currently, I don't see the use case, if needed in some edge case it can be fetched using the OMP/CUDA API.
If needed we can also have a member variable associated with the executor, storing the upper limit, every time `bulk_execute` is called that variable gets set.

Reply:
253ed92 Should resolve this